File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        map<int,int>m;
        int n=nums.size();
        vector<int>ans;
        for(int i{};i<n;i++)
        {
            if(m.find(target-nums[i])!=m.end())
            {
                ans.push_back(i);
                ans.push_back(m[target-nums[i]]);
            }
            m[nums[i]]=i;
        }
        return ans;
        
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict={}
        for ind,ele in enumerate(nums):
            diff=target-ele
            if diff in dict:
                return [dict[diff],ind]
            else:
                dict[ele]=ind
            